unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+import uuid
+from unittest import mock
+
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import patch
+
+area_id = str(uuid.uuid4())
+area_uid = None
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
+
+
+@pytest.fixture
+def redis_client_auto_register():
+    return RedisClientBase(area_id=area_id, autoregister=False)
+
+
+class TestRedisClientBase:
+    @classmethod
+    def setup_class(cls):
+        cls.obj = RedisClientBase(area_id=area_id, autoregister=False)
+
+    def test_constructor_registry_success(self, pubsub_thread=None):
+        assert self.obj.area_id == area_id
+        assert self.obj.area_uuid == area_uid
+        assert self.obj.is_active == False
+        assert self.obj._subscribed_aggregator_response_cb is None
+        assert self.obj._subscribe_to_response_channels(pubsub_thread) is None
+        assert self.obj._blocking_command_responses == {}
+        assert self.obj._transaction_id_buffer == []
+
+    def test_function_subscribe_to_response_channels_success(self):
+        RedisClientBase._subscribe_to_response_channels = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        test._subscribe_to_response_channels.assert_called()
+        test.pubsub.psubscribe.assert_called()
+
+    def test_function_aggregator_response_callback_success(self):
+        RedisClientBase._aggregator_response_callback = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        test._aggregator_response_callback()
+        test._aggregator_response_callback.assert_called_once()
+
+    @pytest.fixture()
+    def register_fixture(self):
+        initial = RedisClientBase.register
+        RedisClientBase.register = MagicMock()
+        test = RedisClientBase(area_id=area_id, autoregister=False)
+        yield test
+        RedisClientBase.register = initial
```

Comment:
I think that these tests can be omitted. You are basically doing this:
- replace `RedisClientBase.register()` with a mock;
- call `RedisClientBase.register()`;
- assert that `RedisClientBase.register()` was called.
But this will of course pass, and it doesn't tell us anything about the state of the actual implementation (because we're basically testing the mock!). As Hannes said, this is not exactly a unit-test. I think the same applies to the other fixtures below.